StoneMasonKarel.java

```diff
@@ -1,17 +1,69 @@
-/*
- * File: StoneMasonKarel.java
- * --------------------------
- * The StoneMasonKarel subclass as it appears here does nothing.
- * When you finish writing it, it should solve the "repair the quad"
- * problem from Assignment 1.  In addition to editing the program,
- * you should be sure to edit this comment so that it no longer
- * indicates that the program does nothing.
- */
-
-import stanford.karel.*;
-
-public class StoneMasonKarel extends SuperKarel {
-
-	// You fill in this part
-
-}
+/*
+ * File: StoneMasonKarel.java
+ * --------------------------
+ * The StoneMasonKarel subclass as it appears here does nothing.
+ * When you finish writing it, it should solve the "repair the quad"
+ * problem from Assignment 1.  In addition to editing the program,
+ * you should be sure to edit this comment so that it no longer
+ * indicates that the program does nothing.
+ */
+
+import stanford.karel.*;
+

```

Comment:
Your solution assumes that you are dealing with a world with four columns to fix that have the same height. However this is not the case. The assignment reads:

"The end of the columns is marked by a wall immediately after the final column. This
wall section appears after 13th Avenue in the example, but your program should work
for any number of columns." and

"The top of the column is marked by a wall, but Karel cannot assume that columns are
always five units high, or even that all columns are the same height."